addons/udes_stock_refactoring/models/stock_picking.py

```diff
@@ -0,0 +1,367 @@
+                be searched and returned. If used in conjunction with
+                priorities then only those pickings of those ids will be
+                returned.
+
+            @param (optional) picking_type_ids: Array (int)
+                If it is set the pickings returned will be only from the picking types in the array.
+        """
+        Picking = self.env["stock.picking"]
+        Package = self.env["stock.quant.package"]
+        Users = self.env["res.users"]
+
+        order = None
+
+        if states is None:
+            states = ["draft", "cancel", "waiting", "confirmed", "assigned", "done"]
+
+        warehouse = Users.get_user_warehouse()
+        if picking_type_ids is None:
+            picking_type_ids = warehouse.get_picking_types().ids
+
+        if self:
+            domain = [("id", "in", self.mapped("id"))]
+        elif origin:
+            domain = [("origin", "=", origin)]
+        elif backorder_id:
+            domain = [("backorder_id", "=", backorder_id)]
+        elif result_package_id:
+            domain = [("move_line_ids.result_package_id", "=", result_package_id)]
+        elif product_id:
+            if not location_id:
+                raise ValidationError(_("Please supply a location_id"))
+            domain = [
+                ("move_line_ids.product_id", "=", product_id),
+                ("move_line_ids.location_id", "=", location_id),
+            ]
+        elif package_name:
+            package = Package.get_package(package_name, no_results=True)
+            if not package:
+                return Picking.browse()
+            domain = self._get_package_search_domain(package)
+        elif picking_priorities:
+            domain = [
+                ("priority", "in", picking_priorities),
+                ("picking_type_id", "=", warehouse.pick_type_id.id),
+                ("batch_id", "=", False),
+            ]
+            if picking_ids is not None:
+                domain.append(("id", "in", picking_ids))
+            order = "priority desc, scheduled_date, id"
+        elif picking_ids:
+            domain = [("id", "in", picking_ids)]
+        elif location_id:
+            warehouse = Users.get_user_warehouse()
+            domain = [
+                ("location_id", "=", location_id),
+                ("picking_type_id", "=", warehouse.int_type_id.id),
+            ]
+        elif batch_id is not None:
+            domain = [("batch_id", "=", batch_id)]
+        else:
+            raise ValidationError(_("No valid options provided."))
+
+        # add the states to the domain
+        domain.append(("state", "in", states))
+        # add the picking type ids to the domain
+        domain.append(("picking_type_id", "in", picking_type_ids))
+
+        # add extra domain if there is any
+        if extra_domain:
+            domain.extend(extra_domain)
+
+        pickings = Picking.search(domain, order=order)
+
+        return pickings
+
+    def action_assign(self):
```

Comment:
I'm wondering if these action functions should also be moved to udes_stock? We may want to remove junk for other situations as well

Reply:
Do we also need a function for action_done to unlink empty? It seems to be done at the end of the UDES11 method

Reply:
Also inconsistent docstring format

Reply:
I've gone back and forth on this a few times. For now I think all unlinking of empty pickings should be done in the stock refactoring module, as right now that is the only time you'd make a picking empty/want to auto delete empty pickings.

Reply:
Have now updated the docstrings.